promql/quantile.go

```diff
@@ -203,14 +203,19 @@ func histogramQuantile(q float64, h *histogram.FloatHistogram) float64 {
 		rank = (1 - q) * h.Count
 	}
 
+	bucketIndex := -1
```

Comment:
was trying to replicate the original logic in bucketQuantile where they check if it's the first/last bucket. added this variable since this one uses an iterator and we can't use the index directly. but you're right that we can probably assume that the bucket with lower -Inf is the first bucket and the bucket with upper +Inf is the last bucket and not bother checking for first/last